prov/usnic/src/usdf_av.c

```diff
@@ -469,6 +473,8 @@ usdf_am_insert_sync(struct fid_av *fav, const void *addr, size_t count,
 
 		if (addr_format_str) {
 			usdf_str_toaddr(addr_str[i], &cur_sin);
+			if (NULL == cur_sin)
+				return -FI_ENOMEM;
```

Comment:
Looking at the man page, I don't think we're supposed to return negative return values for this call. Which of course, means the values returned above are problematic. 

> Insertion calls for an AV opened for synchronous operation will return the
number of addresses that were successfully inserted. In the case of failure,
the return value will be less than the number of addresses that was specified.